src/common/storage/storageclasses/patchstorage.js

```diff
@@ -0,0 +1,41 @@
+/*globals define*/
+/*jshint node:true*/
+/**
+ * @author kecso / https://github.com/kecso
+ */
+define([
+    'common/util/jsonPatcher',
+    'common/storage/constants',
+    'common/storage/storageclasses/objectloaders',
+    'q'
+], function (jsonPatcher, CONSTANTS, StorageObjectLoaders, Q) {
+    'use strict';
+
+    function PatchStorage(webSocket, logger, gmeConfig) {
+        StorageObjectLoaders.call(this, webSocket, logger, gmeConfig);
+        this.logger = this.logger || logger.fork('patch');
+    }
+
+    PatchStorage.prototype = Object.create(StorageObjectLoaders.prototype);
+
+    PatchStorage.prototype.constructor = PatchStorage;
+
+    PatchStorage.prototype._fillPatchRoot = function (commitData, rootInfo) {
+
+        if (this.gmeConfig.storage.patchRootCommunicationEnabled &&
```

Comment:
I think `this.gmeConfig.storage.patchRootCommunicationEnabled` should only be applied at two points.
In the core when returing rootHash/rootInfo and on the server side when dispatching the event.